### DISCUSSION TITLE
Provide builds

### DISCUSSION THREAD

**Discussion by krazyjakee** [godot-jolt/godot-jolt#272] [Questions]
I'm not used to visual studio or C++ build tools. The amount of configuration involved on windows is pretty laughable having followed the instructions AND having predefined configuration files.

I added cmake to my PATH which worked in visual studio but no matter what configuration I chose, it only build the 32 bit debug version. So installed visual studio build tools and ran that console but it's pulling the PATH from somewhere I can't find and it's looking for some executable called "cl" and... frankly I give up.

It would be nice if somebody could zip up some builds even though it's not ready for any production use. That way people can start checking it out and providing feedback without wasting a day figuring out how to build it.

**Comment by mihe** [comment]
Yeah, I understand your frustration. Building other people's C++ projects can be a nightmare for even experienced C++ developers. I struggled with whether I should write the documentation for people who are already familiar with C++/CMake and its complicated conventions, or whether I should write it for people who aren't. In the end I decided I didn't want to take on the latter, since that's almost a project in itself.

If you're okay with waiting a bit, I'm planning to start pushing beta releases within the next 2-3 weeks, assuming no major roadblocks, since I'm more or less done with the first phase of this extension from a feature standpoint.

If not, here's a one-off Windows build for you, including the examples project: _[removed]_

You should be able to drop that "extensions" folder (or just the extension files themselves) into any other project and then be able select "JoltPhysics3D" as your physics engine in your project settings. Just make sure you have "Advanced Settings" enabled, otherwise you won't be able to see the setting. Also, it's probably a good idea to add a "JoltDebugGeometry3D" node to your scene, to verify that you're actually running the correct physics engine.

If that all works, then great, otherwise just hang on for a couple of weeks. I'd prefer to sort out stuff like code signing and other release-related things before I start pushing proper releases to GitHub.

If you feel like giving source builds another attempt, you can try using CMake to generate a proper Visual Studio solution instead. This does however require a regular Visual Studio 2022 installation, with the C++ workload, so "Build Tools for Visual Studio 2022" is not enough.

You don't need any special command prompt for this, just run this from any regular Windows command prompt:

```pwsh
path\to\cmake.exe -S "path\to\godot-jolt" -B "path\to\godot-jolt\build\vs2022" -G "Visual Studio 17 2022"
```

That'll give you a `build\vs2022\godot-jolt.sln` file that you can open. Once in Visual Studio just select "EditorDistribution" from the configuration drop-down in the toolbar and then "Build > Build Solution". That should give you a DLL in the `bin` folder which the examples project can use.

**Comment by krazyjakee** [comment]
This is great thanks!
